### PR TITLE
[Feat]: ViewTab: add editor-link

### PR DESCRIPTION
**Pull request by ferasbbm**
Hi Dear,
i made this PR to add the editor link in view tab, cuz every time I want to open the view file, I copy its path and go to the IDE and find the file name.

Hope it will be accepted soon...

Thanks in advance.

### CODE REVIEW

File: src/DataCollector/ViewCollector.php [hunk 1]
```diff
@@ -5,12 +5,37 @@
...
+        'vscode-remote' => 'vscode://vscode-remote/%file',
+        'vscode-insiders-remote' => 'vscode-insiders://vscode-remote/%file',
+        'vscodium' => 'vscodium://file/%file',
+        'nova' => 'nova://core/open/file?filename=%file',
+        'xdebug' => 'xdebug://%file',
+        'atom' => 'atom://core/open/file?filename=%file',
+        'espresso' => 'x-espresso://open?filepath=%file',
+        'netbeans' => 'netbeans://open/?f=%file',
+    ];
+
```
Comment (ferasbbm) [inline src/DataCollector/ViewCollector.php:38]: add editors-links variable

File: src/DataCollector/ViewCollector.php [hunk 2]
```diff
@@ -44,6 +70,36 @@ public function getWidgets()
...
+            );
+        }
+
+        $filePath = $this->replaceSitesPath($filePath);
+
+        $url = str_replace(['%file'], [$filePath], $this->editors[ config('debugbar.editor') ]);
+
+        return $url;
+    }
+
```
Comment (ferasbbm) [inline src/DataCollector/ViewCollector.php:102]: This function checks if the IDE's URL is available or not, and if it is, it returns the full url of the file in the IDE by merging the editor link and the file link together

File: src/DataCollector/ViewCollector.php [hunk 3]
```diff
@@ -84,6 +141,7 @@ public function addView(View $view)
             'param_count' => count($params),
             'params' => $params,
             'type' => $type,
+            'editorLink' => $editorLink,
```
Comment (ferasbbm) [inline src/DataCollector/ViewCollector.php:144]: return the link to view

File: src/DataCollector/ViewCollector.php [hunk 4]
```diff
@@ -102,4 +160,16 @@ public function collect()
...
+     * Replace remote path
+     *
+     * @param string $filePath
+     *
+     * @return string
+     */
+    protected function replaceSitesPath($filePath)
+    {
+        return str_replace(config('debugbar.remote_sites_path'), config('debugbar.local_sites_path'), $filePath);
+    }
```
Comment (ferasbbm) [inline src/DataCollector/ViewCollector.php:174]: This function for merging the editor link and the file link together

**Review by ferasbbm — COMMENTED** [review]
Add some comments to explain the code

**Comment by parallels999** [comment]
@barryvdh already added on #1359

**Pull request closed without merging**